src/Core/Vipr.Core/OutputFile.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Vipr.Core
+{
+    public abstract class OutputFile
```

Comment:
OutputFile is a bit of a misnomer here since TextFiles which derive from OutputFile are also used on input. How about we call it RelativeFile since its purpose is to name the file relative to an arbitrary/unknown root path?